listenbrainz/webserver/views/playlist_api.py

```diff
@@ -691,3 +691,47 @@ def copy_playlist(playlist_mbid):
         raise APIInternalServerError("Failed to copy the playlist. Please try again.")
 
     return jsonify({'status': 'ok', 'playlist_mbid': new_playlist.mbid})
+
+
+@playlist_api_bp.route("/<playlist_mbid>/export/<service>", methods=["POST", "OPTIONS"])
+@crossdomain
+@ratelimit()
+@api_listenstore_needed
+def export_playlist(playlist_mbid, service):
+    """
+
+    Export a playlist to an external service.
+
+    :reqheader Authorization: Token <user token>
+    :statuscode 200: playlist copied.
+    :statuscode 401: invalid authorization. See error message for details.
+    :statuscode 404: Playlist not found
+    :resheader Content-Type: *application/json*
+    """
+    user = validate_auth_header()
+
+    if not is_valid_uuid(playlist_mbid):
+        log_raise_400("Provided playlist ID is invalid.")
+
+    if service != "spotify":
+        raise APIBadRequest(f"Service name {service} is invalid or currently not supported")
+
+    spotify_service = SpotifyService()
+    token = spotify_service.get_user(user["id"])
+    if not token:
+        raise APIBadRequest(f"Service {service} is not linked. Please link your {service} account first.")
+
+    if not SPOTIFY_LISTEN_PERMISSIONS.issubset(set(token["scopes"])):
+        raise APIBadRequest(f"Missing required scopes to export playlists."
```

Comment:
I think it would be good to state explicitly which scopes, using the exact same terminology at the spotify site uses, the user needs.

Reply:
Makes sense. I updated the text. However, note that currently I perform the same check on frontend side and hide the button entirely if the scopes are absent. Do you think we should do something else on the frontend in that case?